cores/rp2040/CoreMutex.cpp

```diff
@@ -57,7 +56,7 @@ CoreMutex::~CoreMutex() {
         if (__isFreeRTOS) {
             auto m = __get_freertos_mutex_for_ptr(_mutex);
             if (__freertos_check_if_in_isr()) {
-                __freertos_mutex_give_from_isr(m);
+                __freertos_mutex_give_from_isr(m, true);
```

Comment:
Here, return that stored member variable and I think we're good.  This isolates all changes to the CoreMutex object itself, transparently.

Reply:
Changed to use the CoreMutex member variable